src/main/java/com/mineinabyss/mobzy/api/nms/entity/ProjectileHelpers.kt

```diff
@@ -0,0 +1,47 @@
+package com.mineinabyss.mobzy.api.nms.entity
+
+import net.minecraft.server.v1_16_R2.Entity.c
+import net.minecraft.server.v1_16_R2.IProjectile
+import net.minecraft.server.v1_16_R2.MathHelper
+import net.minecraft.server.v1_16_R2.Vec3D
+import org.bukkit.util.Vector
+import kotlin.math.atan2
+import kotlin.math.sqrt
+import kotlin.random.Random
+
+
+/**
+ * Shoot the projectile towards a direction, with a certain speed and randomness
+ * @param dX X direction
+ * @param dY Y direction
+ * @param dZ Z direction
+ * @param speed The speed at which the projectile should travel
+ * @param randomAngle The maximum random angle that can be applied to the projectile shooting direction. Set to 0 to aim directly at the target.
+ */
+//TODO: Implement the entity deltaPosition ticking ourselves at some point, to circumvent NMS.
+fun IProjectile.shootDirection(dX: Double, dY: Double, dZ: Double, speed: Float, randomAngle: Double) {
+    val directionVector = Vector(dX, dY, dZ).normalize()
+    if(randomAngle != 0.0) {
+        directionVector.rotateAroundX(Random.nextDouble(-randomAngle, randomAngle).toRadians())
+        directionVector.rotateAroundY(Random.nextDouble(-randomAngle, randomAngle).toRadians())
+        directionVector.rotateAroundZ(Random.nextDouble(-randomAngle, randomAngle).toRadians())
+        directionVector.normalize()
+    }
+    directionVector.multiply(speed)
+
+    mot = Vec3D(directionVector.x, directionVector.y, directionVector.z)
+
+    // NMS stuff for orienting the projectile model towards the target
+    val horizontalDistanceSqrt = sqrt(directionVector.x * directionVector.x + directionVector.z * directionVector.z)
+    yaw = (atan2(directionVector.x, directionVector.z) * 57.2957763671875).toFloat()
```

Comment:
Maybe make a `const val` instead of repeating the same number so it's clearer what it is. If it's the simplification of some formula involving pi, you could honestly just name it 
```kotlin
var `whatever * the * equation ^ is`
```

If that's just converting radians to degrees, use some built in math function.

Reply:
that float is from NMS code, not sure what it represents but I just copied it over

Reply:
oooh it is for radians to degrees, just found this in some random library when i searched for it `#define | RAD2DEG   57.2957763671875`
I'll change it to use the built in function then yeah